packages/react-magma-dom/src/components/Pagination/SimplePagination.tsx

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import { ButtonColor, ButtonShape, ButtonVariant } from '../Button';
+import { I18nContext } from '../../i18n';
+import { Tooltip } from '../Tooltip';
+import { ArrowBackIcon, ArrowForwardIcon } from 'react-magma-icons';
+import styled from '../../theme/styled';
+import { ThemeContext } from '../../theme/ThemeContext';
+import { NativeSelect } from '../NativeSelect';
+import { NavButton, PaginationProps } from './';
+import { useGenerateId } from '../../utils';
+import { Spacer } from '../Spacer';
+import { VisuallyHidden } from '../VisuallyHidden';
+import { transparentize } from 'polished';
+
+function buildLabelColor(props) {
+  if (props.isInverse) {
+    if (props.disabled) {
+      return transparentize(0.8, props.theme.colors.neutral100);
+    }
+    return props.theme.colors.neutral100;
+  }
+  if (props.disabled) {
+    return props.theme.colors.neutral500;
+  }
+  return props.theme.colors.neutral700;
+}
+
+const StyledWrapper = styled.div<{
+  disabled?: boolean;
+  isInverse?: boolean;
+}>`
+  display: flex;
+  align-items: center;
+  max-width: 360px;
+  select {
+    min-width: 87px;
+  }
+  label {
+    color: ${buildLabelColor};
+    font-size: ${props => props.theme.typeScale.size02.fontSize};
+    font-weight: 500;
+  }
+`;
+
+export const SimplePagination = React.forwardRef<
+  HTMLDivElement,
+  PaginationProps
+>((props, ref) => {
+  const {
+    count,
+    defaultPage,
+    disabled,
+    hideNextButton,
+    hidePreviousButton,
+    id: defaultId,
+    isInverse,
+    testId,
+    onPageChange,
+    ...other
+  } = props;
+  const theme = React.useContext(ThemeContext);
+
+  const i18n = React.useContext(I18nContext);
+
+  const id = useGenerateId(defaultId);
+
+  let [selectedPage, setSelectedPage] = React.useState(
+    defaultPage ? defaultPage : 1
+  );
+
+  function handleChange(event) {
+    setSelectedPage(event.target.value);
+    props.onChange &&
+      typeof props.onChange === 'function' &&
+      props.onChange(event);
+  }
+
+  //Decreases number by one on previous button click
+  function handlePrev() {
+    if (selectedPage > 1) {
+      setSelectedPage(selectedPage - 1);
+    }
+  }
+
+  //Increases number by one on next button click
+  function handleNext() {
+    if (selectedPage < count) {
+      setSelectedPage(++selectedPage);
+    }
+  }
+
+  function paginationLabel() {
+    if (count <= 1) {
+      return `${i18n.simplePagination.ofLabel}
+        ${count}
+        ${i18n.simplePagination.pageLabel}`;
+    } else {
+      return `${i18n.simplePagination.ofLabel}
+        ${count}
+        ${i18n.simplePagination.pagesLabel}`;
+    }
```

Comment:
This works great and you can just leave it. But **if** you want an extra little challenge, can you think of another way to write this with less code repeat? hint: it can be done in a single line :)

Reply:
Accepted!